genie-web/src/main/java/com/netflix/genie/web/configs/GenieAgentFilterAutoConfiguration.java

```diff
@@ -0,0 +1,131 @@
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.configs;
+
+import com.netflix.genie.web.properties.AgentFilterProperties;
+import com.netflix.genie.web.services.AgentFilterService;
+import com.netflix.genie.web.services.impl.AgentFilterServiceImpl;
+import com.netflix.genie.web.util.AgentMetadataInspector;
+import com.netflix.genie.web.util.BlacklistedVersionAgentMetadataInspector;
+import com.netflix.genie.web.util.InspectionReport;
+import com.netflix.genie.web.util.MinimumVersionAgentMetadataInspector;
+import com.netflix.genie.web.util.WhitelistedVersionAgentMetadataInspector;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.Order;
+
+import java.util.List;
+
+/**
+ * Auto-configuration for the {@link AgentFilterService} default implementation.
+ * This component is activated unless another bean is present AND if the corresponding property is set to true.
+ * <p>
+ * This configuration also creates a set of pre-configured {@link AgentMetadataInspector} that the
+ * service loads and uses:
+ * - Version whitelist: accept agent IFF version matches a given pattern
+ * - Version blacklist: reject agent whose version matches a given pattern
+ * - Minimum version: reject agent whose version is lower than a given minimum
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Configuration
+@ConditionalOnProperty(value = AgentFilterProperties.VERSION_FILTER_ENABLED_PROPERTY, havingValue = "true")
+@EnableConfigurationProperties(
+    {
+        AgentFilterProperties.class,
+    }
+)
+public class GenieAgentFilterAutoConfiguration {
+
+    /**
+     * Get a {@link AgentFilterService} instance if there isn't already one.
+     * Annotate inspector beans with {@link Order} if precedence is important.
+     *
+     * @param agentMetadataInspectorsList the list of inspectors.
+     * @return An {@link AgentFilterService} instance.
+     */
+    @Bean
+    public AgentFilterService agentFilterService(
+        final List<AgentMetadataInspector> agentMetadataInspectorsList
+    ) {
+        return new AgentFilterServiceImpl(agentMetadataInspectorsList);
+    }
+
+    /**
+     * Fallback NOOP {@link AgentMetadataInspector} created only if no other
+     * bean of this type is present, to satisfy the dependency expressed by {@link AgentFilterService}.
+     *
+     * @return an noop inspector that always returns CONTINUE
+     */
+    @Bean
+    @ConditionalOnMissingBean(AgentMetadataInspector.class)
```

Comment:
Looking at this configuration class it seems impossible for this to even happen. It's not like your other beans have conditionals on them.

Reply:
Ah, indeed. Forgot to take this out.